sklearn/ensemble/tests/test_forest.py

```diff
@@ -194,15 +196,18 @@ def test_probability():
 def check_importances(X, y, name, criterion):
     ForestEstimator = FOREST_ESTIMATORS[name]
 
-    est = ForestEstimator(n_estimators=20, criterion=criterion,random_state=0)
+    est = ForestEstimator(n_estimators=20, criterion=criterion,
+                          random_state=0)
     est.fit(X, y)
     importances = est.feature_importances_
     n_important = np.sum(importances > 0.1)
     assert_equal(importances.shape[0], 10)
     assert_equal(n_important, 3)
 
-    X_new = est.transform(X, threshold="mean")
-    assert_less(X_new.shape[1], X.shape[1])
+    clean_warning_registry()
+    with warnings.catch_warnings(record=True) as record:
```

Comment:
Could you please use `assert_warns` here? Apart from anything else, when deprecation concludes, the test will fail until the assertion is removed.